blackboxopt/visualizations/utils.py

```diff
@@ -171,3 +174,45 @@ def mask_pareto_efficient(costs: np.ndarray):
             )  # Keep any point with a lower cost
             is_efficient[i] = True  # And keep self
     return is_efficient
+
+
+def patch_plotly_io_to_html(method):
+    """Patch `plotly.io.to_html` with additional javascript to improve usability.
+
+    Injects `<script>`-tag with content from `to_html_patch.js` at the end of the HTML
+    output. But only, if the chart title starts with "[BBO]" (to minimize side
+    effects, if the user uses `plotly.io` for something else).
+
+    `plotly.io.to_html` is also internally used for `figure.show()` and
+    `figure.to_html()`, so this is covered, too.
+
+    Args:
+        method: Original `plotly.io.to_html` method.
+
+    Returns:
+        Patched method.
+    """
+
+    @wraps(method)
+    def wrapped(*args, **kwargs):
+        html = method(*args, **kwargs)
+
+        # Test if title text contains "[BBO]"
+        if html.find('"title": {"text": "[BBO]') < 0:
```

Comment:
I introduced a convention, that the charts title has to start with `[BBO]`, otherwise the patch is not applied.

I did this to mitigate the conflicting goals of:
- Need to patch `plotly.io.to_html` globally, so our charts receive the patch, no matter how they are generated (`fig.show()` or `fig.to_html()` call `plotly.io.to_html()` internally)
- Minimize the side effects on the users, if they generates other charts with ploty.

What do you think? Any other ideas?